src/tensor/traits.ts

```diff
@@ -61,6 +62,210 @@ export const nullLikeTraitValues = [
+        args.name &&
+        args.currentTraits &&
+        isAlpha(args.name) &&
+        !args.currentTraits.has(SynthTrait.LETTERS_ONLY)
+      ) {
+        return { trait_type: SynthTrait.LETTERS_ONLY, value: "true" };
+      }
+    },
+  },
+  {
+    type: TraitType.DOMAIN,
+    generate: (args: SynthTraitArg) => {
+      if (
+        args.name &&
+        args.currentTraits &&
+        emojiLength(args.name) > 0 &&
+        emojiLength(args.name) === visibleLength(args.name) &&
+        !args.currentTraits.has(SynthTrait.LANGUAGE)
+      ) {
+        return { trait_type: SynthTrait.LANGUAGE, value: SynthTrait.EMOJI };
+      }
+    },
+  },
+  {
+    type: TraitType.DOMAIN,
+    generate: (args: SynthTraitArg) => {
+      if (
+        args.name &&
+        args.currentTraits &&
+        isPalindrome(args.name) &&
+        !args.currentTraits.has(SynthTrait.PALINDROME)
+      ) {
+        return { trait_type: SynthTrait.PALINDROME, value: "true" };
+      }
+    },
+  },
+  {
+    type: TraitType.DOMAIN,
+    generate: (args: SynthTraitArg) => {
+      if (
+        args.name &&
+        args.currentTraits &&
+        visibleLength(args.name) >= 3 &&
+        visibleLength(args.name) <= 5 &&
+        !args.currentTraits.has(`${visibleLength(args.name)} Letters}`)
+      ) {
+        return {
+          trait_type: `${visibleLength(args.name)} Letters`,
+          value: "true",
+        };
+      }
+    },
+  },
+  {
+    type: TraitType.DOMAIN,
+    generate: (args: SynthTraitArg) => {
+      if (args.name && args.currentTraits) {
+        const numberClub = determineNumberClub(args.name);
+        if (numberClub && !args.currentTraits.has(SynthTrait.CATEGORY)) {
+          return {
+            trait_type: SynthTrait.CATEGORY,
+            value: numberClub,
+          };
+        }
+      }
+    },
+  },
+  {
+    type: TraitType.DOMAIN,
+    generate: (args: SynthTraitArg) => {
+      if (args.name && args.currentTraits && !args.currentTraits.has(SynthTrait.LANGUAGE)) {
+        for (const [language, expression] of Object.entries(languageRegex)) {
+          if (expression.test(args.name) && emojiLength(args.name) == 0 ) {
+            return {
+              trait_type: SynthTrait.LANGUAGE,
+              value: language,
+            };
+          }
+        }
+      }
+    },
+  },
+];
+
+
+const createSyntheticTraits = (
+  args: SynthTraitArg,
+  traitType: TraitType,
+  currentAttributes: Attribute[]
+) => {
+  let newAttributes: Attribute[] = [];
+  const { name } = args;
+
+  // Knowing there's no standard, formatting like this reduces chance of duplicates in the future
+  const currentTraits = new Set(currentAttributes.map((trait) => trait.trait_type));
+  for (const synthTrait of syntheticTraits) {
+    if (synthTrait.type == traitType) {
+      const newTrait = synthTrait.generate({ name, currentTraits });
+      if (newTrait) {
+        newAttributes.push(newTrait);
+      }
+    }
+  }
+  return newAttributes;
+};
```

Comment:
Might be good to add a couple of unit tests for each type

Reply:
yup! I have them in tensor-infra for parseExt, but i might as well just move it over